src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java

```diff
@@ -403,8 +409,9 @@ private ResponseEntity<PostePayAuthResponse> createPostePayAuthResponse(String c
         postePayAuthResponse.setChannel(channel);
         postePayAuthResponse.setRequestId(requestId);
         postePayAuthResponse.setCorrelationId(correlationId);
+        String sessionToken = jwtTokenUtils.generateToken(requestId);
         if (StringUtils.isEmpty(errorMessage)) {
-            String urlRedirect = String.format(pgsResponseUrlRedirect, requestId, Scopes.POSTEPAY_SCOPE);
+            String urlRedirect = StringUtils.join(String.format(pgsResponseUrlRedirect, requestId, Scopes.POSTEPAY_SCOPE), "#token=", sessionToken);
```

Comment:
If you expect this strings to be not null prefer using `+` to concatenate strtings

Reply:
Done https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/210/commits/2ea78190d0495b5dfadfdafe1a469312ce6be9a0